File: app/src/main/java/com/o4x/musical/ui/dialogs/SetTagsDialog.java
package com.o4x.musical.ui.dialogs;

import android.app.Dialog;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.fragment.app.DialogFragment;

import com.afollestad.materialdialogs.MaterialDialog;
import com.o4x.musical.R;

public class SetTagsDialog extends DialogFragment {

    private final On on;

    public SetTagsDialog(On on) {
        this.on = on;
    }

    public static class On {
        protected void allTags() {};
        protected void justImage() {};
    }


    public static SetTagsDialog create(On on) {
        return new SetTagsDialog(on);
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        MaterialDialog.Builder builder =
                new MaterialDialog.Builder(getActivity())
                        .title(R.string.download_tags)
                        .content(R.string.download_tags_message)
                        .neutralText(android.R.string.cancel)
                        .negativeText(R.string.just_image)
                        .positiveText(R.string.all_tags)
                        .onNeutral((dialog, which) -> {})
                        .onNegative((dialog, which) -> on.justImage())
                        .onPositive((dialog, which) -> on.allTags());
        return builder.build();
    }
}

File: app/src/main/java/com/o4x/musical/ui/dialogs/SetTagsDialog.kt
package com.o4x.musical.ui.dialogs

import android.app.Dialog
import android.os.Bundle
import androidx.fragment.app.DialogFragment
import com.afollestad.materialdialogs.MaterialDialog
import com.o4x.musical.R

class SetTagsDialog(private val on: On) : DialogFragment() {
    open class On {
        open fun allTags() {}
        open fun justImage() {}
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {

        return MaterialDialog(requireContext())
            .title(R.string.download_tags)
            .message(R.string.download_tags_message)
            .neutralButton(R.string.cancel)
            .negativeButton(R.string.just_image) {
                on.justImage()
            }
            .positiveButton(R.string.all_tags) {
                on.allTags()
            }
    }

    companion object {
        @JvmStatic
        fun create(on: On): SetTagsDialog {
            return SetTagsDialog(on)
        }
    }
}